cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -251,26 +250,27 @@ func (p *Planner) categorizeNodes(podDestinations map[string]bool, scaleDownCand
 	}
 	p.nodeUtilizationMap = utilizationMap
 	timer := time.NewTimer(p.context.ScaleDownSimulationTimeout)
+RemovalSimulation:
 	for i, node := range currentlyUnneededNodeNames {
 		select {
 		case <-timer.C:
 			klog.Warningf("%d out of %d nodes skipped in scale down simulation due to timeout.", len(currentlyUnneededNodeNames)-i, len(currentlyUnneededNodeNames))
-			break
+			break RemovalSimulation
```

Comment:
WDYT about testing if the timeout is actually honored?

Reply:
That's require a timer mock, because with ScaleDownSimulationTimeout=0seconds, the channel still sends a signal later then the loop processing for small number of nodes. 
I don't think we want to move timer to categorizeNodes() variables. Do you have other ideas how to mock the timer?

Reply:
I was thinking rather about using real timer, but mocking removalSimulator to make sure it is not called after the timeout.